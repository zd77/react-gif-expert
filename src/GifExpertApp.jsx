import React, { Fragment, useState } from 'react'
import { AddCategory, GifGrid } from './components/'

export const GifExpertApp = () => {
    const [ categories, setCategories ] = useState(['Naruto'])
    const onAddCategory = ( newCategory ) => {
        if( categories.includes( newCategory ) ) return
        setCategories( [ newCategory , ...categories] )
    }
    return (
    <Fragment>
        <h1>GifExpertApp</h1>
        <AddCategory 
            // setCategories={ setCategories }
            onNewCategory={ event => onAddCategory( event ) }
        />
        {
            categories.map(( category ) => (
                <GifGrid key={ category } category={ category } /> 
            ))
        }
    </Fragment>
    )
}
