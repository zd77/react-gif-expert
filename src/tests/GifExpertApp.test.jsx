import { render, screen, fireEvent } from '@testing-library/react'
import { GifExpertApp } from '../GifExpertApp'

describe('Tests on <GifExpertApp />', () => { 
    test('should match with the snapshot', () => { 
        const { container } = render( <GifExpertApp /> )
        expect( container ).toMatchSnapshot()
     })
     test('should not add a repeated category', () => { 
        const category = 'SamuraiX'
        render( <GifExpertApp /> )
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input( input, { target: { value: category }} )
        fireEvent.submit( form )
        fireEvent.input( input, { target: { value: category }} )
        fireEvent.submit( form )
        screen.debug()
        expect( screen.getAllByRole('heading', { level: 3 } ).length ).toBe( 1 )
      })
 })