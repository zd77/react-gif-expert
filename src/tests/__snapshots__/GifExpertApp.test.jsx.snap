// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> should match with the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search for GIFs"
      type="text"
      value=""
    />
  </form>
</div>
`;
